homeassistant/components/remote/lg_netcast.py

```diff
@@ -0,0 +1,116 @@
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_ACCESS_TOKEN):
+        vol.All(cv.string, vol.Length(max=6)),
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the LG Netcast remotes."""
+
+    from pylgnetcast import LgNetCastClient
+    host = config[CONF_HOST]
+    access_token = config.get(CONF_ACCESS_TOKEN)
+    name = config[CONF_NAME]
+
+    client = LgNetCastClient(host, access_token)
+
+    add_entities([LGNetcastRemote(client, name)], True)
+
+
+class LGNetcastRemote(RemoteDevice):
+    """Representation of a LG Netcast remote."""
+
+    def __init__(self, client, name):
+        """Initialize the LG Netcast Remote."""
+        self._client = client
+        self._name = name
+        self._state = False
+        self._last_command_sent = None
+
+    @util.Throttle(MIN_TIME_BETWEEN_SCANS, MIN_TIME_BETWEEN_FORCED_SCANS)
+    def update(self):
+        """Retrieve the latest data from the LG TV."""
+        from pylgnetcast import LgNetCastError, LG_QUERY
+        try:
+            with self._client as client:
+                volume_info = client.query_data(LG_QUERY.VOLUME_INFO)
+                if volume_info:
+                    self._state = STATE_ON
+                else:
+                    self._state = STATE_OFF
+        except (LgNetCastError, RequestException):
+            self._state = STATE_OFF
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if remote is on."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return device state attributes."""
+        if self._last_command_sent is not None:
+            return {'last_command_sent': self._last_command_sent}
+        return None
+
+    def turn_on(self, **kwargs):
+        """Turn the remote on."""
+        self._state = STATE_ON  # Turn on not support by LG Netcast TV
+        self.schedule_update_ha_state()
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the remote off."""
+        await self.async_send_command(1)
+
+    async def async_send_command(self, command, **kwargs):
+        """Send a command to a device."""
+        from pylgnetcast import LgNetCastError, LG_COMMAND
+        num_repeats = kwargs.get(ATTR_NUM_REPEATS)
+
+        delay = kwargs.get(ATTR_DELAY_SECS, DEFAULT_DELAY_SECS)
+        for _ in range(num_repeats):
+            cmdline = ''
+            for cmdline in command:
+                try:
+                    with self._client as client:
+                        cmdline = getattr(LG_COMMAND, cmdline.upper())
+                        client.send_command(cmdline)
+                except (LgNetCastError, RequestException):
+                    self._state = STATE_OFF
+            await asyncio.sleep(delay)
+            self._last_command_sent = cmdline
+        self.schedule_update_ha_state()
```

Comment:
Use async version in async context.
```suggestion
        self.async_schedule_update_ha_state()
```